app/src/main/java/org/systers/mentorship/viewmodels/HomeViewModel.kt

```diff
@@ -0,0 +1,79 @@
+package org.systers.mentorship.viewmodels
+
+import android.arch.lifecycle.LiveData
+import android.arch.lifecycle.MutableLiveData
+import android.arch.lifecycle.ViewModel
+import android.util.Log
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.disposables.CompositeDisposable
+import io.reactivex.rxkotlin.addTo
+import io.reactivex.schedulers.Schedulers
+import org.systers.mentorship.MentorshipApplication
+import org.systers.mentorship.R
+import org.systers.mentorship.models.HomeStatistics
+import org.systers.mentorship.remote.datamanager.UserDataManager
+import org.systers.mentorship.utils.CommonUtils
+import org.systers.mentorship.utils.SingleLiveEvent
+import retrofit2.HttpException
+import java.io.IOException
+import java.util.concurrent.TimeoutException
+
+
+/**
+ * This class represents the ViewModel
+ * for the HomeFragment
+ */
+class HomeViewModel : ViewModel() {
+
+    private val TAG = this::class.java.simpleName
+    private val userDataManager by lazy { UserDataManager() }
+    private val compositeDisposable by lazy { CompositeDisposable() }
+
+    private val _userStats = MutableLiveData<HomeStatistics>()
+    private val _message = SingleLiveEvent<String>()
+
+    val userStats: LiveData<HomeStatistics>
+        get() = _userStats
+
+    val message: LiveData<String>
+        get() = _message
+
+    init {
```

Comment:
Can you explain why call the function inside `init {}` instead of a function, like other ViewModels? @haroldadmin

Reply:
Yes, sure. I had thought about its implementation in other ways, but I ultimately landed on this one because:

The home fragment is the first fragment that is visible to the users when they open the app. This network call should therefore be made as early as possible to ensure that the data is fetched and displayed at the earliest possible time. Therefore, instead of waiting for the Fragment to complete its initialization, I make the network call as soon as the ViewModel is constructed. This way, we can save some time.

I hope this makes it a little clearer, but if you still suggest I move the call from an init block to a function, I can do that too. What do you think?